### PR TITLE
Mistral AWQ with memory profiling

### PR DESCRIPTION
**Pull request by Anindyadeep**
This PR introduces all the changes by PR #167 and integrates those in AWQ. AWQ README now has quality checks table for both Llama 2 and Mistral

### CODE REVIEW

**Comment by Anindyadeep** [comment]
For the `docs/llama2.md.template`, I added an initial draft on the numbers, lemme know if that looks or not, or we can discuss on this.

**Review by nsosio — COMMENTED** [review]
@Anindyadeep let's merge before PR #160 into main and then merge main into this branch so that I check only the AWQ changes

**Comment by Anindyadeep** [comment]
> @Anindyadeep let's merge before PR #160 into main and then merge main into this branch so that I check only the AWQ changes

Yes that works

File: bench_autoawq/README.md [hunk 1]
```diff
@@ -6,31 +6,56 @@
 
 [AutoAWQ](https://github.com/casper-hansen/AutoAWQ) is a package that is a polished implemementation of the original work [llm-awq](https://github.com/mit-han-lab/llm-awq) from MIT. AWQ or Activation Aware Quantization is a quantization method which supports 4-bit quantization. It massively increases the inference throughput and decreases the memory requirement of the model at the same time. (For example, according to this [reference](https://huggingface.co/TheBloke/Llama-2-70B-Chat-AWQ), Llama2 70B requires 2 x 80 GB but with AutoAWQ it can be run on 1 x 48 GB GPU). You can learn more about AWQ on the research paper and the github implementations.
 
-### 🚀 Running the AutoAWQ Benchmark.
+## 🚀 Running the AutoAWQ Benchmark.
 
-You can run the AutoAWQ benchmark using the following command:
+We can run the AutoAWQ benchmark for two models: [Llama2](https://huggingface.co/meta-llama/Llama-2-7b) and [Mistral-7B v0.1](https://huggingface.co/mistralai/Mistral-7B-v0.1) Here is how we run benchmark for AutoAWQ.
```
Comment (nsosio) [inline bench_autoawq/README.md:11]: are these models correct? Shouldn't they be the chat models?

File: bench_autoawq/README.md [hunk 2]
```diff
@@ -6,31 +6,56 @@
...
+
+## 👀 Some points to note:
 
 1. AutoAWQ is not supported devices other than GPU (only supports when CUDA is available).
 2. We are independently benchmarking AutoAWQ (i.e. the actual AWQ quantization method here). We are not benchmarking with combinations like: AutoAWQ + VLLM or AutoAWQ + TensorRT.
 3. For doing this benchmark, the default model that was choosen was: [Llama2-AutoAWQ by The Bloke](https://huggingface.co/TheBloke/Llama-2-7B-AWQ)
-4. AutoAWQ does not support INT8 quantization properly yet. See [this issue](https://github.com/casper-hansen/AutoAWQ/issues/45).
+4. AutoAWQ does not support INT8 quantization properly yet. See [this issue](https://github.com/casper-hansen/AutoAWQ/issues)
```
Comment (nsosio) [inline bench_autoawq/README.md:61]: the issue has been closed?

File: bench_autoawq/bench.sh [hunk 3]
```diff
@@ -137,12 +124,8 @@ while [ "$#" -gt 0 ]; do
...
             shift 2
             ;;
-        -lf|--log_file)
-            LOG_FILENAME="$2"
-            shift 2
-            ;;
-        -md|--models_dir)
-            MODELS_DIR="$2"
+        -n|--model_name)
+            MODEL_NAME="$2"
```
Comment (nsosio) [inline bench_autoawq/bench.sh:128]: add check to model name

**Review by nsosio — COMMENTED** [review]
@Anindyadeep can we merge before the [PR #167](https://github.com/premAI-io/benchmarks/pull/167) and merge dev into branch so that I only check the engine changes? Same for the other PRs.

File: bench_autoawq/README.md (hunk 2 above)
Comment (Anindyadeep) [reply bench_autoawq/README.md]: Actually this still remains incomplete from this PR: https://github.com/casper-hansen/AutoAWQ/pull/71

File: bench_autoawq/bench.sh (hunk 3 above)
Comment (Anindyadeep) [reply bench_autoawq/bench.sh:128]: This check is applied in base.py

File: bench_autoawq/README.md (hunk 1 above)
Comment (Anindyadeep) [reply bench_autoawq/README.md]: Changed the model names with the links

**Review by nsosio — APPROVED** [review]

**Pull request merged**